tutorials/plot_spatial.py

```diff
@@ -11,40 +11,215 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Use x- and y-position to compute distance between two or more points, cumulative distance 
+# traveled and speed
+#
+
+###################################################################################################
+
+# Look at the range of our x and y positions
+ranges = get_pos_ranges(position)
+print(f'The x-position ranges from {ranges[0][0]} to {ranges[0][1]}')
+print(f'The y-position ranges from {ranges[1][0]} to {ranges[1][1]}')
+
+###################################################################################################
+
+# Plot positions (coordinates marked x are the actual points)
+plot_positions(position, alpha = 1, ls = '-', marker = 'x', c = 'tab:gray', markersize = 10)
+plt.title('Tracking')
+plt.xlabel('x-position')
+plt.ylabel('y-position')
+plt.legend(['coordinates'])
+
+###################################################################################################
+
+# Compute distance between start and end point
+dist_start_end = compute_distance(x_pos[0], y_pos[0], x_pos[-1], y_pos[-1])
+# Compute distance traveled at each point
+dist_traveled = compute_distances(x_pos, y_pos)
+# Compute total distance traveled
+cumulative_dist_traveled = compute_cumulative_distances(x_pos, y_pos)
+# Compute speed at each point
+bin_widths = np.array([1, 1, 1, 2, 1.5, 1, 1, 1, 2, 1, 1, 1, 1, 1, 0.5])
+speeds = compute_speed(x_pos, y_pos, bin_widths)
+
+###################################################################################################
+
+f, (ax1, ax2, ax3) = plt.subplots(3, 1, sharex = True)
+# plot distance traveled at each time
+plot_positions(np.append([x_pos[1:]], [dist_traveled], axis = 0), 
+               ax = ax1, alpha = 1, ls = '-', marker = 'x', c = 'tab:pink', markersize = 10)
+ax1.set_title('Distance traveled at each point')
+ax1.set_xlabel('time (t)')
+ax1.set_ylabel('distance (u)')
+
+# plot cumulative distance traveled per time
+plot_positions(np.append([x_pos[1:]], [cumulative_dist_traveled], axis = 0),
+               ax = ax2, alpha = 1, ls = '-', marker = 'x', c = 'tab:olive', markersize = 10)
+ax2.set_title('Cumulative distance traveled at each point')
+ax2.set_xlabel('time (t)')
+ax2.set_ylabel('distance (u)')
+
+# plot speed at each time point
+plot_positions(np.append([x_pos[1:]], [speeds], axis = 0),
+               ax = ax3, alpha = 1, ls = '-', marker = 'x', c = 'tab:cyan', markersize = 10)
+ax3.set_title('Speed at each point')
+ax3.set_xlabel('time (t)')
+ax3.set_ylabel('speed (u/t)')
+
+# add some padding between subplots
+f.tight_layout(pad = 0.05)
+# make figure bigger
+f.set_size_inches((15/2.54, 20/2.54))
+
+###################################################################################################
+# 2. Divide position in spatial bin edges and plot
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Compute x- and y- spatial bin edges and plot spatial grid. 
```

Comment:
Here I think we want to add a little bit more information on what we mean by binning, and why we want to do this. 

Can you add a draft of a sentence or two on this here?